packages/angular_devkit/build_angular/src/builders/jest/test-files.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { IOptions as GlobOptions, glob as globCb } from 'glob';
+import { promisify } from 'util';
+import { JestBuilderOptions } from './options';
+
+const globAsync = promisify(globCb);
+
+/**
+ * Finds all test files in the project.
+ *
+ * @param options The builder options describing where to find tests.
+ * @param workspaceRoot The path to the root directory of the workspace.
+ * @param glob A promisified implementation of the `glob` module. Only intended for
+ *     testing purposes.
+ * @returns A set of all test files in the project.
+ */
+export async function findTestFiles(
```

Comment:
This is a good idea, and I hadn't considered it earlier. Thinking about it now however, I'm inclined _not_ to reuse Karma's test discovery logic for a few reasons:

1. Direct 1:1 compatibility between Jest and Karma is currently not a goal.
2. I personally think some of the behavior in `findTests` is not intuitive or necessarily helpful. Directory inclusion and `.spec` injection both sound like features I would not want to include in the first place unless compatibility was an objective.
3. Karma's implementation hard-codes `.spec` as a convention in both of the above features, even if projects are actually using a different convention like `.test`.
4. The behavior of `include` is dependent on the actual file system layout which I think is a bit of an anti-pattern for a feature like this.
5. The `test-files.ts` file is currently a bit coupled to Webpack. It doesn't look that hard to separate out but we generally can't assume a dependency on Webpack or Karma exists at all and need to make sure the import graph aligns with that expectation.

I'm certainly open to revisiting if users really do like those extra features or get tripped up by the difference with the existing Karma builder, but I don't think it's strictly necessary for this initial experiment.

Reply:
Thinking about it a bit more, the `include` karma functionality was mainly intended to match Jest features, but with Jest this might be a bit redundant, since Jest can already handle most of this features such as running all tests in a specific directory.

Reply:
Interesting, I wasn't aware of that history. For now, let's keep `include` functionality simple, and we can always expand it later if there's enough desire for it.